### PR TITLE
Chore: use github app for issue commands workflow

### PR DESCRIPTION
**Pull request by academo**
**What is this feature?**

Replaces the usage of a github access token with an github app generated one in the issues command workflow.

**Why do we need this feature?**

We were constantly hitting the rate limit of the github api when running the issues command workflow, this was likely due to the
token being reused in other workflows.

**Who is this feature for?**

All issues created in the grafana/grafana repository.

### CODE REVIEW

File: .github/workflows/issue-opened.yml [hunk 1]
```diff
@@ -29,22 +64,8 @@ jobs:
         uses: ./actions/commands
         with:
           metricsWriteAPIKey: ${{secrets.GRAFANA_MISC_STATS_API_KEY}}
-          token: ${{secrets.ISSUE_COMMANDS_TOKEN}}
+          token: ${{ steps.generate_token.outputs.token }}
           configPath: "issue-opened"
-  config:
```
Comment (academo) [inline .github/workflows/issue-opened.yml:34]: move to the top

**Review by tolzhabayev — APPROVED** [review]

File: .github/workflows/issue-opened.yml [hunk 2]
```diff
@@ -10,9 +15,42 @@ permissions:
...
+  config:
+    runs-on: "ubuntu-latest"
+    outputs:
+      has-secrets: ${{ steps.check.outputs.has-secrets }}
+    steps:
+      - name: "Check for secrets"
+        id: check
+        shell: bash
+        run: |
+          if [ -n "${{ (secrets.GRAFANA_DELIVERY_BOT_APP_ID != '' &&
```
Comment (tolzhabayev) [inline .github/workflows/issue-opened.yml:27]: why the check for those secrets is still needed?

Comment (academo) [reply .github/workflows/issue-opened.yml:27]: I'll change them to check for the repository name

File: .github/workflows/issue-opened.yml [hunk 3]
```diff
@@ -10,61 +15,86 @@ permissions:
...
+        shell: bash
+        run: |
+          if [ "${{ github.repository }}" == "grafana/grafana" ]; then
+            echo "has-secrets=1" >> "$GITHUB_OUTPUT"
+          fi
+
   main:
     runs-on: ubuntu-latest
+    needs: config
+    if: needs.config.outputs.has-secrets
```
Comment (tolzhabayev) [inline .github/workflows/issue-opened.yml:34]:
```suggestion
    if: github.repository == 'grafana/grafana'
```

I don't see any reason to spawn a "config" job at all if you can just do this check directly. Just add this check and remove the whole `config` job above.

Comment (academo) [reply .github/workflows/issue-opened.yml:34]: Good idea!

**Review by tolzhabayev — APPROVED** [review]

**Pull request merged**